examples/example3/nautobot_adapter.py

```diff
@@ -0,0 +1,167 @@
+        try:
+            country = diffsync.nautobot.dcim.regions.create(
+                slug=ids.get("slug"),
+                name=attrs.get("name"),
+                description=attrs.get("subregion", None),
+                parent=region.remote_id,
+            )
+            print(f"Created country : {ids} | {attrs} | {country.id}")
+
+        except pynautobot.core.query.RequestError as exc:
+            print(f"Unable to create country {ids} | {attrs} | {exc}")
+            return None
+
+        # Add the newly created remote_id and create the internal object for this resource.
+        attrs["remote_id"] = country.id
+        item = super().create(ids=ids, diffsync=diffsync, attrs=attrs)
+        return item
+
+    def update(self, attrs: dict):
+        """Update a country object in Nautobot.
+
+        Args:
+            attrs: Dictionary of attributes to update on the object
+
+        Returns:
+            DiffSyncModel: this instance, if all data was successfully updated.
+            None: if data updates failed in such a way that child objects of this model should not be modified.
+
+        Raises:
+            ObjectNotUpdated: if an error occurred.
+        """
+
+        # Retrive the pynautobot object from Nautobot since we only have the UUID internally
+        remote = self.diffsync.nautobot.dcim.regions.get(self.remote_id)
+
+        # Convert the internal attrs to Nautobot format
+        nautobot_attrs = {}
+        if "subregion" in attrs:
+            nautobot_attrs["description"] = attrs.get("subregion")
+        if "name" in attrs:
+            nautobot_attrs["name"] = attrs.get("name")
+
+        if nautobot_attrs:
+            remote.update(data=nautobot_attrs)
+            print(f"Updated Country {self.slug} | {attrs}")
+
+        return super().update(attrs)
+
+    def delete(self):
+        """Delete a country object in Nautobot.
+
+        Returns:
+            NautobotCountry: DiffSync object
+        """
+        # Retrieve the pynautobot object and delete the object in Nautobot
+        remote = self.diffsync.nautobot.dcim.regions.get(self.remote_id)
+        remote.delete()
+
+        super().delete()
+        return self
+
+
+class NautobotAdapter(DiffSync):
+    """Example of a DiffSync adapter implementation."""
+
+    # We are using NautobotCountry and NautobotRegion instead of Region and Country
+    # because we are using these classes to manage the logic to integrate with Nautobot
+    # NautobotRegion is just a small extension to store the UUID and do not support any CRUD operation toward Nautobot
+    # NautobotCountry support the creation, update or deletion of a country in Nautobot
+    region = NautobotRegion
+    country = NautobotCountry
+
+    # Since all countries are associated with a region, we don't need to list country here
+    # When doing a diff or a sync between 2 adapters,
+    #  diffsync will recursively check all models defined at the top level and their children.
```

Comment:
```suggestion
    # When doing a diff or a sync between 2 adapters,
    #  diffsync will recursively check all models defined at the top level and their children.
    # Since countries are defined as children of a region, we don't need to list country here
```